content/en/blog/_posts/2021-07-21-memory-manager-beta.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: blog
+title: "Kubernetes Memory manager moves to beta"
+date: 2021-07-21
+slug: kubernetes-1-22-feature-topoloy-manager-beta
+---
+
+# Memory manager moves to beta
+
+The blog post describes the internals of the memory manager, a beta feature of Kubernetes 1.22. The **Memory Manager** is a component in the **kubelet** ecosystem proposed to enable the feature of guaranteed memory (and hugepages) allocation for pods in the Guaranteed QoS class.
+
+This blog post covers:
+
+1. Why do you need it?
+2. The internal details of how the **MemoryManager** works
+3. Current limitations of the **MemoryManager**
+4. Future work for the **MemoryManager**
+
+## Why do you need it?
+
+To get the best performance and latency for the workload container, container CPUs, PCI devices, and memory should be aligned to the same NUMA node. The kubelet already provides a set of managers to align CPUs and PCI devices, but you did not have a way to align memory. By default, the kernel will try to allocate the memory from the same NUMA node where container CPUs are placed, but it can not guarantee it.
+
+## How does it work?
+
+The memory manager is doing two main things:
+- provides the topology hint to the topology manager
+- allocates the memory for containers and updates the state
+
+The overall sequence of the Memory Manager
```

Comment:
Are these part of scheduling? (no) How can we help make sure the reader is clear on when these decisions are taken?